clients/drcachesim/tracer/tracer.cpp

```diff
@@ -3216,17 +3235,19 @@ drmemtrace_client_main(client_id_t id, int argc, const char *argv[])
         /* we use placement new for better isolation */
         DR_ASSERT(MAX_INSTRU_SIZE >= sizeof(offline_instru_t));
         placement = dr_global_alloc(MAX_INSTRU_SIZE);
-        instru = new (placement) offline_instru_t(
-            insert_load_buf_ptr, op_L0_filter.get_value(), &scratch_reserve_vec,
-            file_ops_func.write_file, module_file, op_disable_optimizations.get_value());
+        instru = new (placement)
+            offline_instru_t(insert_load_buf_ptr, op_L0I_filter.get_value(),
+                             &scratch_reserve_vec, file_ops_func.write_file, module_file,
+                             op_disable_optimizations.get_value() ||
+                                 op_L0D_filter.get_value() || op_L0I_filter.get_value());
```

Comment:
-use_physical sets disable_optimizations to true: should filtering do the same?  Then the file type would have the no-opt flag set (which might simplify raw2trace logic?).

Reply:
Done.

Reply:
I was able to simply the raw2trace logic just a bit. In `analyze_elidable_addresses`, now I don't need to check `OFFLINE_FILE_TYPE_IFILTERED` and `OFFLINE_FILE_TYPE_DFILTERED`. I still check `OFFLINE_FILE_TYPE_FILTERED` as that's a legacy type. 

I still need to check the new offline file types for instrs_is_separate and the assert statement for `memref.use_remembered_base`.